mod_reforged/hooks/config/rumors.nut

```diff
@@ -0,0 +1,68 @@
+// We add up 4 new variables for texts to be build with.
+// Three are wrong distances, direction and terrain for the purpose of creating more interesting rumors.
+// One is an indirect adjective for a legendary location if you don't wanna name it directly
+// This hook will add those vars for contracts and events aswell but they will not make use of those
+local buildTextFromTemplate = ::buildTextFromTemplate;
+::buildTextFromTemplate = function( _text, _vars )
+{
+    foreach (var in _vars)
+    {
+        if (var[0] == "distance")
+        {
+            local wrongDistances = [];
+            foreach (distance in ::Const.Strings.Distance) if (distance != var[1]) wrongDistances.push(distance);
+            _vars.push([
+                "wrongDistance",
+                wrongDistances[::Math.rand(0, wrongDistances.len() - 1)]
+            ]);
+            continue;
+        }
+        if (var[0] == "direction")
+        {
+            local wrongDirections = [];
+            foreach (direction in ::Const.Strings.Direction8) if (direction != var[1]) wrongDirections.push(direction);
+            _vars.push([
+                "wrongDirection",
+                wrongDirections[::Math.rand(0, wrongDirections.len() - 1)]
+            ]);
+            continue;
+        }
+        if (var[0] == "terrain")
+        {
+            local wrongTerrains = [];
+            foreach (terrain in ::Const.Strings.Terrain) if (terrain != var[1] && terrain != "") wrongTerrains.push(terrain);
+            _vars.push([
+                "wrongTerrain",
+                wrongTerrains[::Math.rand(0, wrongTerrains.len() - 1)]
+            ]);
+            continue;
+        }
+    }
+    _vars.push([
+        "legendaryLocationAdjective",
+        ::Const.Strings.LegendaryLocationAdjective[::Math.rand(0, ::Const.Strings.LegendaryLocationAdjective.len() - 1)]
+    ])
+    return buildTextFromTemplate(_text, _vars);
+}
+
+::Const.Strings.LegendaryLocationAdjective <- [
+	"overwhelming",
+	"mystical",
+	"mindblowing",
+	"never before seen",
+	"spectacular",
+    "surreal"
+];
+
+::Const.Strings.RumorsUniqueLocation <- [
+    [   // Doesn't include the name but the information is correct
+        "{I keep hearing talk | Some traveller mentioned something | Some pilgrim mentioned something} about a %legendaryLocationAdjective% {place | location} {%direction% from here | %terrain% | %distance%}"
+    ],
+    [   // Include the exact name but the other information is usually randomly true or fake
+        "{An explorer | A cartographer} just recently came by, said he found {a | some} %legendaryLocationAdjective% {place | location} %distance% {that he | } {called | named} %location%. If only some more details slipped out of his mouth...",
```

Comment:
Something's wrong with `{that he | }`

Reply:
Oh that's on purpose. I want to increase the variety of these messages so I add these branches whenever possible and it makes sense.

Either it's a location `... that he called Witch Hut ... `
Or it's a location `... called Witch Hut ...`